src/plugins/kibana_usage_collection/server/collectors/application_usage/telemetry_application_usage_collector.ts

```diff
@@ -48,9 +70,31 @@ export interface ApplicationUsageTelemetryReport {
     minutes_on_screen_7_days: number;
     minutes_on_screen_30_days: number;
     minutes_on_screen_90_days: number;
+    views?: ApplicationViewUsage[];
   };
 }
 
+export const transformByApplicationViews = (
+  report: ApplicationUsageViews
+): ApplicationUsageTelemetryReport => {
+  const reportMetrics = Object.values(report);
+  const mainApplications = reportMetrics.filter(
+    (appView) => appView.viewId === MAIN_APP_DEFAULT_VIEW_ID
+  );
+  const appViews = reportMetrics.filter((appView) => appView.viewId !== MAIN_APP_DEFAULT_VIEW_ID);
+
+  return mainApplications.reduce((acc, mainApplication) => {
+    const currentAppViews = appViews.filter((appView) => appView.appId === mainApplication.appId);
+
+    acc[mainApplication.appId] = {
+      ...mainApplication,
+      views: currentAppViews,
+    };
+
+    return acc;
```

Comment:
Super-NIT: AFAIK, the whole point of `reduce` vs. a `for` loop is to not mutate the `acc` object but return a new object instead:

```suggestion
    return {
      ...acc,
      [mainApplication.appId]: {
        ...mainApplication,
        views: currentAppViews,
      }
    };
```

Reply:
True. I think it is fine since we are not mutating an object outisde the scope of the reduce function. Creating a new object on every iteration has performance costs and is not as readable in some cases. I am fine with either in most cases.